src/Microsoft.Security.Utilities.Core/HighPerformanceScanner.cs

```diff
@@ -0,0 +1,505 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Security.Utilities;
+
+#if NET8_0_OR_GREATER
+using System.Buffers;
+using StringInput = System.ReadOnlySpan<char>;
+#else
+using StringInput = string;
```

Comment:
>StringInput [](http://example.com/codeflow?start=6&length=11)

just asking the question, if we took a conditional dependency on System.Buffers for the netstandard build, would our ability to consume ReadOnlySpan<char> provide any worthwhile perf gains?

Reply:
There's no netstandard API to match a regex on a span so I ended up having to do it with indices, so no.